cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/model/ClusterModelStats.java

```diff
@@ -257,42 +248,48 @@ public String toString() {
 
   /**
    * Generate statistics of utilization for resources among alive brokers in the given cluster.
+   * Average and standard deviation calculations are based on brokers not excluded for replica moves.
    *
    * @param clusterModel The state of the cluster.
+   * @param optimizationOptions Options to take into account while retrieving cluster capacity.
    */
-  private void utilizationForResources(ClusterModel clusterModel) {
+  private void utilizationForResources(ClusterModel clusterModel, OptimizationOptions optimizationOptions) {
     // Average, maximum, and standard deviation of utilization by resource.
-    Map<Resource, Double> avgUtilizationByResource = new HashMap<>();
-    Map<Resource, Double> maxUtilizationByResource = new HashMap<>();
-    Map<Resource, Double> minUtilizationByResource = new HashMap<>();
-    Map<Resource, Double> stDevUtilizationByResource = new HashMap<>();
+    Map<Resource, Double> avgUtilizationByResource = new HashMap<>(Resource.cachedValues().size());
+    Map<Resource, Double> maxUtilizationByResource = new HashMap<>(Resource.cachedValues().size());
+    Map<Resource, Double> minUtilizationByResource = new HashMap<>(Resource.cachedValues().size());
+    Map<Resource, Double> stDevUtilizationByResource = new HashMap<>(Resource.cachedValues().size());
     for (Resource resource : Resource.cachedValues()) {
-      double avgUtilizationPercentage = clusterModel.load().expectedUtilizationFor(resource) / clusterModel.capacityFor(resource);
+      double resourceUtilization = clusterModel.load().expectedUtilizationFor(resource);
+      double avgUtilizationPercentage = resourceUtilization / clusterModel.capacityWithAllowedReplicaMovesFor(resource, optimizationOptions);
```

Comment:
Is there any chance for the denominator (return value of the below method) to be 0?
```
clusterModel.capacityWithAllowedReplicaMovesFor(resource, optimizationOptions);
```

Reply:
TL;DR. Yes, and that is expected and acceptable for cluster model stats.

--
Model stats for resource utilization presents `min`, `max`, `std`, and `average`. Division by zero makes the average and the std infinity. This can happen when the cluster has no load to balance. `ExcludedBrokersForReplicaMoveTest` has test cases that triggers this scenario (e.g. please see `tid=2` for resource distribution goals).